### PR TITLE
PrivateRoutes added. History refactored to import

### PR DESCRIPTION
**Pull request by BtheGit**
-I added a PrivateRoute component (per RRv4 docs practices).
-The login page now tracks the previous location and passes that to the login thunk to redirect appropriately. I added a ternary filter to make sure the login page doesn't redirect back to itself.
-Per SOP, to give proper access to history in the action creators, it needed to be separated from the App component.
-I set home and qualified programs as private routes.
-Since we don't have a 404 page I used a redirect bucket at the bottom of the switch to redirect all erroneous requests to the homepage.

### CODE REVIEW

File: app/src/Login.js [hunk 1]
```diff
@@ -96,8 +98,8 @@ const Login = props => {
 
 Login.propTypes = {
   userInput: PropTypes.string,
-  isAuthenticated: PropTypes.bool,
-  isPending: PropTypes.bool,
+  isAuthenticated: PropTypes.bool.isRequired,
+  isPending: PropTypes.bool.isRequired,
   errorMessage: PropTypes.string
```
Comment (mkalish) [inline app/src/Login.js]: Looks like the login function is missing.

**Comment by mkalish** [comment]
Looks like there is an issue if I navigate directly to the login page.

**Comment by BtheGit** [comment]
Good catch on the direct route, ternary fix. The login isn't missing just didn't validate against proptype, fixed that as well.

**Comment by BtheGit** [comment]
And the other action creator function is covered now as well.

**Comment by mkalish** [comment]
@BtheGit 
Looks awesome, can you take a look at the conflict when you get a chance and we'll get this merged?

**Pull request merged**